meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit.bb

```diff
@@ -24,8 +25,10 @@ inherit deploy
 do_configure[noexec] = "1"
 
 do_compile() {
+	sed -e '/@@INCLUDE_COMMON@@/ {' -e 'r ${S}/boot.cmd.in' -e 'd' -e '}' \
+			"${S}/boot.cmd" > boot.cmd.in
 	sed -e 's/@@FIT_NODE_SEPARATOR@@/${FIT_NODE_SEPARATOR}/g' \
-			"${S}/boot.cmd" > boot.cmd
+			boot.cmd.in > boot.cmd
```

Comment:
We can probably have one sed with multiple commands here, which would be easier to understand what is being replaced.

Reply:
yes, but will new command use the result of the previous command in this case ?( if we decide to use `-i` option and then multiple commands `-e CMD1 -e CMD2 -e CMD3`)

Reply:
Yeah, that might be tricky. Not a sed expert here, so whatever works best for you it is fine by me :-)

Reply:
it can be all done in one sed, and you can use ; instead of -e

Reply:
I would make is simpler, I will make 2 files and cat them together into sed

Reply:
@Tim-Anderson 
I don't want to add `boot-common.cmd.in` to the end of custom `boot.cmd`, but replace `@@INCLUDE_COMMON@@` in custom `boot.cmd` with the contents of `template.boot.cmd`. So basically mimic the behavior of `#include <file>`, as it is possible to have custom instructions after that include. 

@ricardosalveti A also tried to do that, unfortunately all that commands are independent, next command doesn't use results of the previous one 
```
 sed -e '/@@INCLUDE_COMMON@@/ {' -e 'r ${S}/boot-common.cmd.in' -e 'd' -e '}' \
            -e 's/@@FIT_NODE_SEPARATOR@@/${FIT_NODE_SEPARATOR}/g' \             
                      "${S}/boot.cmd" > boot.cmd                              

```

So I ended up with this result:
```
# set default fdt_file
run findfdt
echo "Using ${fdt_file}"

# Default boot type and device
setenv bootlimit 3
setenv devtype mmc
setenv devnum ${mmcdev}
setenv rootpart 2

setenv fdt_addr 85800000
setenv optee_ovl_addr 0x86000000

# common boot handlers

setenv bootcmd_resetvars 'setenv kernel_image; setenv bootargs; setenv kernel_image2; setenv bootargs2'
setenv bootcmd_otenv 'run bootcmd_resetvars; ext4load ${devtype} ${devnum}:2 ${loadaddr} /boot/loader/uEnv.txt; env import -t ${loadaddr} ${filesize}'
setenv bootcmd_load_f 'ext4load ${devtype} ${devnum}:${rootpart} ${loadaddr} "/boot"${kernel_image}'
setenv bootcmd_tee_ovy 'imxtract ${loadaddr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file} fdt@@FIT_NODE_SEPARATOR@@${fdt_file} ${fdt_addr}; fdt addr ${fdt_addr}; fdt resize 0x1000; fdt apply ${optee_ovl_addr}'
setenv bootcmd_run 'bootm ${loadaddr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file} ${loadaddr}#conf@@FIT_NODE_SEPARATOR@@${fdt_file} ${fdt_addr}'
setenv bootcmd_rollbackenv 'setenv kernel_image ${kernel_image2}; setenv bootargs ${bootargs2}'
setenv bootcmd_set_rollback 'if test ! "${rollback}" = "1"; then setenv rollback 1; setenv upgrade_available 0; saveenv; fi'
setenv bootostree 'run bootcmd_load_f; run bootcmd_tee_ovy; run bootcmd_run'
setenv altbootcmd 'run bootcmd_otenv; run bootcmd_set_rollback; if test -n "${kernel_image2}"; then run bootcmd_rollbackenv; fi; run bootostree; reset'

if test ! -e ${devtype} ${devnum}:1 uboot.env; then saveenv; fi

if test "${rollback}" = "1"; then run altbootcmd; else run bootcmd_otenv; run bootostree; if test ! "${upgrade_available}" = "1"; then setenv upgrade_available 1; saveenv; fi; reset; fi

reset
```

Reply:
@ricardosalveti @Tim-Anderson let me know if you're ok about keeping that as it is (based on results of my previous test)

Reply:
if it works I am okay with it